syntax = "proto3";

package adlik.serving;
option cc_enable_arenas = true;

import "adlik_serving/framework/domain/model_spec.proto";
import "google/protobuf/any.proto";

message CreateTaskRequest {
  ModelSpec model_spec = 1;

  enum TaskType {
    NONE_TASK = 0;
    TRAINING_TASK = 1;
    PREDICT_TASK = 2;
  }

  TaskType task_type = 2;
  bool is_sync = 3;

  // information for specific task, different model has own definition
  google.protobuf.Any detail = 4;
}

message CreateTaskResponse {
  string status = 1; // NEW, RUNNING, ERROR, DONE

  string task_id = 2;

  // information for specific task, different model has own definition
  google.protobuf.Any detail = 3;
}

message QueryTaskRequest { string task_id = 1; }

message QueryTaskResponse { string status = 1; }
